src/backend/distributed/commands/role.c

```diff
@@ -275,3 +514,236 @@ makeDefElemInt(char *name, int value)
+			{
+				arg = makeFloatConst(configurationValue, -1);
+				break;
+			}
+
+			case PGC_BOOL:
+			case PGC_STRING:
+			case PGC_ENUM:
+			{
+				arg = makeStringConst(configurationValue, -1);
+				break;
+			}
+
+			default:
+			{
+				ereport(ERROR, (errmsg("Unrecognized run-time parameter type for %s",
+									   configurationName)));
+				break;
+			}
+		}
+	}
+	else
+	{
+		arg = makeStringConst(configurationValue, -1);
+	}
+	return (Node *) arg;
+}
+
+
+/*
+ * makeStringConst creates a Const Node that stores a given string
+ *
+ * copied from backend/parser/gram.c
+ */
+static Node *
+makeStringConst(char *str, int location)
+{
+	A_Const *n = makeNode(A_Const);
+
+	n->val.type = T_String;
+	n->val.val.str = str;
+	n->location = location;
+
+	return (Node *) n;
+}
+
+
+/*
+ * makeIntConst creates a Const Node that stores a given integer
+ *
+ * copied from backend/parser/gram.c
+ */
+static Node *
+makeIntConst(int val, int location)
+{
+	A_Const *n = makeNode(A_Const);
+
+	n->val.type = T_Integer;
+	n->val.val.ival = val;
+	n->location = location;
+
+	return (Node *) n;
+}
+
+
+/*
+ * makeIntConst creates a Const Node that stores a given Float
+ *
+ * copied from backend/parser/gram.c
+ */
+static Node *
+makeFloatConst(char *str, int location)
+{
+	A_Const *n = makeNode(A_Const);
+
+	n->val.type = T_Float;
+	n->val.val.str = str;
+	n->location = location;
+
+	return (Node *) n;
+}
+
+
+/*
+ * ConfigGenericNameCompare compares two config_generic structs based on their
+ * name fields. If the name fields contain the same strings two structs are
+ * considered to be equal.
+ *
+ * copied from guc_var_compare in utils/misc/guc.c
+ */
+static int
+ConfigGenericNameCompare(const void *a, const void *b)
+{
+	const struct config_generic *confa = *(struct config_generic *const *) a;
+	const struct config_generic *confb = *(struct config_generic *const *) b;
+	return pg_strcasecmp(confa->name, confb->name);
+}
+
+
+/*
+ * ShouldPropagateAlterRoleSetQueries decides if an AlterRoleSetStmt should be
+ * propagated to worker nodes;
+ */
+static bool
+ShouldPropagateAlterRoleSetQueries(HeapTuple tuple, TupleDesc
+								   DbRoleSettingDescription)
+{
+	if (!ShouldPropagate())
+	{
+		return false;
+	}
+
+	const char *currentDatabaseName = CurrentDatabaseName();
+	const char *databaseName =
+		GetDatabaseNameFromDbRoleSetting(tuple, DbRoleSettingDescription);
+	const char *roleName = GetRoleNameFromDbRoleSetting(tuple, DbRoleSettingDescription);
+
+	/*
+	 * session defaults for databases other than the current one are not propagated
+	 */
+	if (databaseName != NULL &&
+		pg_strcasecmp(databaseName, currentDatabaseName) != 0)
+	{
+		ereport(NOTICE, (errmsg("Citus partially supports ALTER ROLE .. IN DATABASE"
+								" .. SET  for distributed databases"),
+						 errdetail("Citus propagates the session defaults that affect "
+								   "current database"),
+						 errhint("You can manually change attributes of roles "
+								 "on workers")));
```

Comment:
Don't know if we need to have this notice, but if we do, I expect it at most once during an add node, and not for every setting for a different database we find as is currently the case.

I opt for removing the notice